### PR TITLE
Forms

### PR DESCRIPTION
**Pull request by chimwemwe007**
## The following has been accomplished in this milestone

- Create a method called current_user in your ApplicationController to make current user data available to all controllers.
It will return the first user from the database.
-  Don't worry about proper authentication at this time. You will add it in the upcoming projects.
Create forms to perform the following functions:
 - Creates a Post on behalf of the current_user (use the method that you created in your ApplicationController).
 - Create a comment on behalf of the current_user (use the method that you created in your ApplicationController).
Allow Users to add likes to Posts.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -3,7 +3,8 @@
 # Rails Blog App
 
 > The Blog app is Roby on rails website that will show the list of posts and empower readers to interact with them by adding comments and liking posts.
-> In this project for the blog app i had to change the validation for the differentmodels as well as writeunit tests for them.
+> In this project for the blog app i had to change the validation for the different models as well as write unit tests for them. Also i had to include forms for the different models.
+
```
Comment (elisha2kyakpo1) [inline README.md]: - [ ] Kindly list down all the commands required to run the project on the user's local machine, eg, `bundle install`, `rails db:create`, etc.

**Review by elisha2kyakpo1 — CHANGES_REQUESTED** [review]
Hi 👋 @chimwemwe007, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights

- The PR title and description text looks good and professional👍
- No linter errors 👍
- Good user of gitflow 👍


## Required Changes ♻️
_Check the comments under the review._

- [x] I couldn't run the project on my machine due to the error in the image below.

migration


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md (hunk 1 above)
Comment (chimwemwe007) [reply README.md]: Hello @elisha2kyakpo1  all changes resuested have been resolved

**Review by Munsa1 — CHANGES_REQUESTED** [review]
Hi @chimwemwe007, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
:heavy_check_mark: Descriptive Readme
:heavy_check_mark: Passing Linters
:heavy_check_mark: Correct Gitflow

 ## Required Changes ♻️
_Check the comments under the review._
## Due to insufficient files in your `files changed` section on github I will leave my comment here
- [ ] As requested by the previous reviewer, I am unable to run the project due to multiple migration files having the same name. Kindly fix this issue.

image


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Munsa1 — CHANGES_REQUESTED** [review]
Hi @chimwemwe007, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
:heavy_check_mark: Descriptive Readme
:heavy_check_mark: Passing Linters
:heavy_check_mark: Correct Gitflow

 ## Required Changes ♻️
_Check the comments under the review._
- [ ] You have not implemened any required changes from me or the first reviewer, please note that that the required changes have to be made. Kindly do so.
image


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/posts_controller.rb [hunk 2]
```diff
@@ -1,11 +1,35 @@
...
+    respond_to do |format|
+      format.html do
+        if @post.save
+          redirect_to user_post_path(current_user, @post), notice: 'Post was successfully created.'
+          # redirect_to '/users/#{@post_new.user_id}/posts', messaage: 'Post was successfully created.'
+        else
+          render :new, status: 'Error: Post was not created.'
+        end
+      end
+    end
```
Comment (jaflih) [inline app/controllers/posts_controller.rb:27]: You have a bug in your project when I create a new post

image


**Follow my others comments to correct it.**

File: db/migrate/20221025191034_create_users.rb [hunk 3]
```diff
@@ -1,9 +1,9 @@
-class CreateUser < ActiveRecord::Migration[7.0]
+class CreateUsers < ActiveRecord::Migration[7.0]
   def change
     create_table :users do |t|
       t.string :name
-      t.string :photo
-      t.string :bio
+      t.text :photo
+      t.text :bio
       t.integer :posts_counter
```
Comment (jaflih) [inline db/migrate/20221025191034_create_users.rb:7]: Please correct this line like this to remove the bug

` t.integer :posts_counter, default: 0`

File: db/migrate/20221025191732_create_posts.rb [hunk 4]
```diff
@@ -1,8 +1,8 @@
-class CreatePost < ActiveRecord::Migration[7.0]
+class CreatePosts < ActiveRecord::Migration[7.0]
   def change
     create_table :posts do |t|
       t.string :title
-      t.text :text
+      t.string :text
       t.integer :comments_counter
```
Comment (jaflih) [inline db/migrate/20221025191732_create_posts.rb:6]: Please correct this line like this to remove the bug

`t.integer :comments_counter, default: 0`

File: db/migrate/20221025191732_create_posts.rb [hunk 5]
```diff
@@ -1,8 +1,8 @@
...
       t.integer :likes_counter
```
Comment (jaflih) [inline db/migrate/20221025191732_create_posts.rb:7]: Please correct this line like this to remove the bug

`t.integer :likes_counter, default: 0`

**Review by jaflih — COMMENTED** [review]
Hi @chimwemwe007 , 

**Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.**

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
The PR title and description text looks good and professional👍
No linter errors 👍
Good user of gitflow 👍

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @jaflih in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request closed without merging**